experimental/rsocket-src/RSocketRequester.cpp

```diff
@@ -37,10 +37,40 @@ RSocketRequester::~RSocketRequester() {
   LOG(INFO) << "RSocketRequester => destroy";
 }
 
-std::shared_ptr<Subscriber<Payload>> RSocketRequester::requestChannel(
-    std::shared_ptr<Subscriber<Payload>> responseSink) {
-  // TODO need to runInEventBaseThread like other request methods
-  return reactiveSocket_->requestChannel(std::move(responseSink));
+yarpl::Reference<yarpl::flowable::Flowable<reactivesocket::Payload>>
+RSocketRequester::requestChannel(
+    yarpl::Reference<yarpl::flowable::Flowable<reactivesocket::Payload>>
+        payloads) {
+  auto& eb = eventBase_;
+  auto srs = reactiveSocket_;
+
+    LOG(INFO) << "requestChannel executing ";
+
+  return yarpl::flowable::Flowables::fromPublisher<Payload>([
+    &eb,
```

Comment:
I am suggesting to capture eb = &eventBase_ which will capture pointer (by value). I also dislike capturing this because it brings questions about lifetime. Capturing pointer to EventBase should be fine since all executors should outlive everything else.